### PR TITLE
don't move to servlet 3.1 automatically

### LINKED ISSUE #1951: Adding <runtime>java8</runtime> auto-updates Dynamic Web Module facet to 3.1, but web.xml remains 2.5
For native Java7 App Engine projects, adding `<runtime>java8</runtime>` in `appengine-web.xml` will auto-magically update the Dynamic Web Module facet and the `server-api` JAR to 3.1 (and the Java facet too), which is nice.

However, `web.xml` still retains 2.5:

```xml
<web-app ...
         xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
         version="2.5">
```

### PR DESCRIPTION
**Pull request by elharo**
fix #1951

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilderTest.java [hunk 1]
```diff
@@ -90,13 +90,13 @@ public void testAddingJava8Runtime() throws CoreException {
     AppEngineDescriptorTransform.addJava8Runtime(appEngineWebDescriptor);
     ProjectUtils.waitForProjects(testProject.getProject());
     assertTrue(testProject.getFacetedProject().hasProjectFacet(JavaFacet.VERSION_1_8));
-    assertTrue(testProject.getFacetedProject().hasProjectFacet(WebFacetUtils.WEB_31));
+    assertTrue(testProject.getFacetedProject().hasProjectFacet(WebFacetUtils.WEB_25));
 
     AppEngineDescriptorTransform.removeJava8Runtime(appEngineWebDescriptor);
     ProjectUtils.waitForProjects(testProject.getProject());
     assertTrue(testProject.getFacetedProject().hasProjectFacet(JavaFacet.VERSION_1_7));
     // can't downgrade from jst.web 3.1
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilderTest.java]: Need to remove this comment.

File: plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilder.java [hunk 2]
```diff
@@ -88,7 +87,6 @@ private void checkRuntimeElement(IFacetedProject project, IFile appEngineWebDesc
         // Can upgrade jst.web to 3.1, but cannot downgrade from 3.1
```
Comment (chanseokoh) [inline plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilder.java]: ditto

File: plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilderTest.java (hunk 1 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.standard.java8.test/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilderTest.java]: done

File: plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilder.java (hunk 2 above)
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilder.java]: done

**Pull request merged**